mne_bids/read.py

```diff
@@ -155,11 +155,14 @@ def _read_events(events_data, event_id, raw, verbose=None):
         verbose=verbose
     )
     all_dur = raw.annotations.duration
-    if all_events.size == 0 and 'rest' not in raw.filenames[0]:
-        warn('No events found or provided. Please add annotations '
-             'to the raw data, or provide the events_data and '
-             'event_id parameters. If this is resting state data '
-             'it is recommended to name the task "rest".')
+
+    # warn no events if not rest
+    if (all_events.size == 0 and hasattr(raw, 'filenames') and raw.filenames[0]
+            is not None and '_task-rest' not in raw.filenames[0]):
```

Comment:
Can this not be simplified to
```suggestion
    if (all_events.size == 0 and  '_task-rest' not in raw.filenames[0]):
```
?

@agramfort

Reply:
No, this function is still called when `filenames` may not exist (preloaded). It is necessary to check whether it exists before checking for "task-rest".

Reply:
Actually, I just realized checking `raw.filenames` doesn't make sense, since the function shouldn't care what people name their files before BIDS conversion.

I will just catch this warning outside the function after checking `bids_path` task for "rest".

Reply:
resolved with #838